### PR TITLE
ksm dedupe

### PR DESCRIPTION
**Pull request by jessegoodier**

### CODE REVIEW

File: custom-tsdb/README.md [hunk 1]
```diff
@@ -54,3 +54,85 @@ Long-term, Kubecost recommends storing the configuration in `cloud-integration.j
 - [AWS](../aws/)
 - [Azure](../azure/)
 - [GCP](../gcp/)
+
+## List of Metrics
+
```
Comment (thomasvn) [inline custom-tsdb/README.md:59]: I think showing the query used to obtain this list would be super helpful!

```
group by(__name__) ({__name__!="", job="kubecost-byo-prom-cost-analyzer"})
```

Where `job` is the name of the scrapeconfig you have configured for Prometheus

File: custom-tsdb/values-primary.yaml [hunk 2]
```diff
@@ -2,14 +2,23 @@ kubecostMetrics:
   # Guide: https://docs.kubecost.com/architecture/ksm-metrics#disabling-ksm-emission
   emitKsmV1Metrics: false # emit all KSM metrics in KSM v1 if true.
   emitKsmV1MetricsOnly: true # emit only the KSM metrics missing from KSM v2, this is required to be true.
+kubecostProductConfigs:
```
Comment (thomasvn) [inline custom-tsdb/values-primary.yaml:5]: Add additional comment here?

```yaml
# drop additional metrics which are still emitted when "emitKsmV1MetricsOnly"
```

**Comment by thomasvn** [comment]
@jessegoodier Made minor comments above for more documentation ^^. Aside from that this looks great!

Should we submit a Github Issue referencing this PR? Proposing that the `emitKSMV1MetricsOnly: true` config by default does not emit the metrics we are dropping here?

**Comment by jessegoodier** [comment]
> @jessegoodier Made minor comments above for more documentation ^^. Aside from that this looks great!
> 
> Should we submit a Github Issue referencing this PR? Proposing that the `emitKSMV1MetricsOnly: true` config by default does not emit the metrics we are dropping here?

we found that a customer implemented the above and was missing:

```
kube_node_labels
```

Would love to figure out why. I think updating the helm chart `values.yaml` and docs are the next steps.

But yes, what you have suggested would be ideal.

**Comment by thomasvn** [comment]
@jessegoodier Does this mean the user needs to remove `kube_node_labels` from the list of `kubecostProductConfigs.metricsConfigs.disabledMetrics`?

**Comment by jessegoodier** [comment]
> @jessegoodier Does this mean the user needs to remove `kube_node_labels` from the list of `kubecostProductConfigs.metricsConfigs.disabledMetrics`?

yes. 

which is great that it is flexible.

I double checked my KSM build and it has that metric. Perhaps they brought it back in the latest KSM. 

Good news is my script worked!

**Review by thomasvn — APPROVED** [review]

File: custom-tsdb/values-primary.yaml [hunk 3]
```diff
@@ -2,14 +2,23 @@ kubecostMetrics:
...
+      - kube_node_labels
+      - kube_node_status_condition
+      - kube_namespace_labels
+      - kube_pod_labels
+      - kube_pod_owner
+  clusterName: kubecostProductConfigs_clusterName # set this to the name of the primary cluster. ideally the same as the value that promClusterIDLabel would be
 kubecostModel:
   containerStatsEnabled: true
   cloudCost:
     enabled: true # Set to true to enable CloudCost view that gives you visibility of your Cloud provider resources cost
```
Comment (srpomeroy) [inline custom-tsdb/values-primary.yaml:19]: Unrelated to this PR, but cloudCost supports OOC.

File: custom-tsdb/values-secondary.yaml [hunk 4]
```diff
@@ -2,15 +2,24 @@ kubecostMetrics:
   # Guide: https://docs.kubecost.com/architecture/ksm-metrics#disabling-ksm-emission
   emitKsmV1Metrics: false # emit all KSM metrics in KSM v1 if true.
   emitKsmV1MetricsOnly: true # emit only the KSM metrics missing from KSM v2, this is required to be true.
+kubecostProductConfigs:
+  metricsConfigs:
+    disabledMetrics: # drop additional metrics which are still emitted when "emitKsmV1MetricsOnly"
```
Comment (srpomeroy) [inline custom-tsdb/values-secondary.yaml:7]: Are these metrics that are deprecated in v2?

Comment (jessegoodier) [reply custom-tsdb/values-secondary.yaml:7]: yes, only deprecated and still present.

**Comment by AjayTripathy** [comment]
https://github.com/kubernetes/kube-state-metrics/issues/1501

@jessegoodier if you're using BYO you need to allow * on kube_pod_labels and kube_node_labels on the metrics labels allowlist...that's why it exists for yours but maybe not others.

**Pull request merged**